graph-viz/app/controllers/GraphController.java

```diff
@@ -15,73 +23,490 @@
+        ArrayList<EdgeVector> dataEdges = new ArrayList<>();
+        ArrayList<Integer> closeEdgeList = new ArrayList<>();
+        for (Map.Entry<Edge, Integer> entry : edges.entrySet()) {
+            Edge edge = entry.getKey();
+            if (sqDistance(edge.getFromLongitude(), edge.getFromLatitude(), edge.getToLongitude(), edge.getToLatitude()) <= 0.001)
+                continue;
+            dataNodes.add(new Point(edge.getFromLongitude(), edge.getFromLatitude()));
+            dataNodes.add(new Point(edge.getToLongitude(), edge.getToLatitude()));
+            dataEdges.add(new EdgeVector(dataNodes.size() - 2, dataNodes.size() - 1));
+            closeEdgeList.add(entry.getValue());
+        }
+        ForceBundling forceBundling = new ForceBundling(dataNodes, dataEdges);
+        forceBundling.setS(zoom);
+        ArrayList<Path> pathResult = forceBundling.forceBundle();
+        int isolatedEdgesCnt = forceBundling.getIsolatedEdgesCnt();
+        ArrayNode pathJson = objectMapper.createArrayNode();
+        int edgeNum = 0;
+        for (Path path : pathResult) {
+            for (int j = 0; j < path.getAlv().size() - 1; j++) {
+                ObjectNode lineNode = objectMapper.createObjectNode();
+                ArrayNode fromArray = objectMapper.createArrayNode();
+                fromArray.add(path.getAlv().get(j).getX());
+                fromArray.add(path.getAlv().get(j).getY());
+                ArrayNode toArray = objectMapper.createArrayNode();
+                toArray.add(path.getAlv().get(j + 1).getX());
+                toArray.add(path.getAlv().get(j + 1).getY());
+                lineNode.putArray("from").addAll(fromArray);
+                lineNode.putArray("to").addAll(toArray);
+                lineNode.put("width", closeEdgeList.get(edgeNum));
+                pathJson.add(lineNode);
+            }
+            edgeNum++;
+        }
+        json = pathJson.toString();
+        objectNode.put("data", json);
+        objectNode.put("isolatedEdgesCnt", isolatedEdgesCnt);
+    }
+
+    private void noBundling(ObjectNode objectNode, HashMap<Edge, Integer> edges) {
+        String json;
+        ArrayNode arrayNode = objectMapper.createArrayNode();
+        for (Map.Entry<Edge, Integer> entry : edges.entrySet()) {
+            ObjectNode lineNode = objectMapper.createObjectNode();
+            lineNode.putArray("from").add(entry.getKey().getFromLongitude()).add(entry.getKey().getFromLatitude());
+            lineNode.putArray("to").add(entry.getKey().getToLongitude()).add(entry.getKey().getToLatitude());
+            lineNode.put("width", entry.getValue());
+            arrayNode.add(lineNode);
+        }
+        json = arrayNode.toString();
+        objectNode.put("data", json);
+        objectNode.put("isolatedEdgesCnt", edges.size());
+    }
+
+    private double sqDistance(double x1, double y1, double x2, double y2) {
+        return Math.pow(x1 - x2, 2) + Math.pow(y1 - y2, 2);
```

Comment:
Is this square distance? then why not do Math.sqrt? Also why not move the function distanceTo from Cluster class to Point class and use it here since the points would be defined from Point class